### PR TITLE
Release 0.16.0

### LINKED ISSUE #681: LDAP 0.16.0
### Reason

- [x] https://github.com/owncloud/user_ldap/pull/678
- [x] https://github.com/owncloud/user_ldap/pull/672
- [x] https://github.com/owncloud/user_ldap/pull/670
- [x] https://github.com/owncloud/user_ldap/pull/671
- [x] https://github.com/owncloud/user_ldap/pull/680
- [x]  https://github.com/owncloud/user_ldap/pull/683
- [x] https://github.com/owncloud/user_ldap/pull/697

### Product approval 

- [ ] Verify transifex commits are in master: If a folder https://github.com/owncloud/user_ldap/tree/master/l10n exists, then https://drone.owncloud.com/owncloud/user_ldap should receive nightly '[tx] updated from transifex` ...
- [ ] Ping pmaier to check `info.xml` description and screenshot
- [ ] Get approval on translations from pmaier - send the date of the last tx commit from drone to Patrick

### QA

- [x] Create **QA** Milestone and set the description to $version
- [x] Move tickets from milestone `development` to `qa` (testplan and tickets mentioned in release ticket)
- [ ] Create release branch with the format "release-$version"
- [x] Create "$version Testplan" ([QA templates](https://github.com/owncloud/QA/tree/master/Server) or [Enterprise](https://github.com/owncloud/qa-enterprise/tree/master/Core)) #682
- [ ] copy new/reopened issues from the previous release ticket as 'Issues to review' below.
- [ ] check if [Non-normal Release Process](https://confluence.owncloud.com/display/OG/Apps+with+differences+in+release+process) applies, and adapt this checklist.

### Documentation

- [ ] Changes in the docs needed? If so create issue in owncloud/docs.

### Marketing

- [ ] Ping @owncloud/marketing
- [x] Update https://confluence.owncloud.com/display/PROD/Upcoming+Releases
- [ ] Request a screenshot if the app is new or changed fundamentally

### Build

All actions to be done on the release branch from here:
- [x] Bump version in `info.xml` (no version suffix)
- [x] Bump "sonar.projectVersion=" in `sonar-project.properties` (no version suffix)
- [x] Check php version dependency in `info.xml`
- [x] Check minimum ownCloud version dependency in `info.xml`
- [x] Change CHANGELOG.md 'Unreleased' to '$version' (today's date)

### Documentation

- [x] Changes in the docs needed? If so create issue in owncloud/docs. -> ttps://github.com/owncloud/docs/issues/3684

### Marketing

- [ ] Ping @owncloud/marketing
- [x] Update https://confluence.owncloud.com/display/PROD/Upcoming+Releases
- [x] Request a screenshot if the app is new or changed fundamentally

#### Beta/RC

- [x] Ping product owner to create github tag for `v${version}-rc1` on release branch (see https://confluence.owncloud.com/display/EN/Product+Owners+List)
- [x] Build and sign RC tarball from github tag `v${version}-rc1` (see handbook for how to build)
- [x] Upload as Github release using changelog entries
- [x] Manual explorative testing:
    - [x] Changelog testing
    - [x] Test and document in testplan => Final "Go" from QA (set 'QA-ready' label) => else do another RC after fixing issues in the "release-$version" branch

#### Final

- [ ] Check that no documentation issues are pending
- [x] Rename last RC tarball to final
- [x] Create github tag for final `v${version}` on release branch
- [x] Upload as Github release using changelog entries
- [ ] Ping `#marketing` channel to coordinate publishing

### Publishing

- [x] Upload tarball to marketplace (see handbook in confluence for details)
- [x] Publish tarball for customers  - [old 10](https://customer.owncloud.com/owncloud/index.php/apps/files/?dir=%2FownCloud-Enterprise%2FownCloud-10), [old 10.0](https://customer.owncloud.com/owncloud/index.php/apps/files/?dir=%2FownCloud-Server%2FownCloud-10.0) and [new portal](https://portal.owncloud.com/apps/files/?dir=/Portal%20Data/All%20Account%20Data/ownCloud%20Enterprise%20Resources%20Data/Server&fileid=5661) (see [handbook](https://confluence.owncloud.com/display/OG/Releasing+Apps) for details)
- [ ] Send mail to release-coordination with link to this ticket and to the changelog
- [ ] Copy email to rocket chat `#updates` channel

### Post-release

- [x] Add a new 'Unreleased' section in CHANGELOG.md (and also update links at the bottom!)
- [x] Create PR to merge changes back to the `master` branch with description "Closes #XYZ" with the release ticket number
- [x] Rename QA milestone to `$version` and close the milestone
- [x] Update https://confluence.owncloud.com/display/PROD/Upcoming+Releases
- [x] Ensure [release template](https://github.com/owncloud/administration-internal/edit/master/tools/release/templates/server_app_release_template.md) is up to date
- [x] Ensure the testplan template ([community](https://github.com/owncloud/QA/tree/master/Server) or [enterprise](https://github.com/owncloud/qa-enterprise/tree/master/Core)) is up to date
- [x] Close the testplan issue

### PR DESCRIPTION
**Pull request by jvillafanez**
Merge release branch into master

Note: I had to rebase, so commits ids are different. The official released content is still accessible via the v0.16.0 tag
Note2: The languages fixes weren't accessible from the v0.16.0 tag so the won't appear if the app is built from the v0.16.0 tag. Anyway, those changes are part of the rebased branch and will be merged into master.

Closes #681

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>core-cliAll-2-1-master-mysql8.0-php7.3</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/user_ldap/3704/92/1

**Comment by jnweiger** [comment]
Sorry for the long delay. Looks good to me. Happy to merge.
Not sure if we need to rebase again before merging...

**Review by jnweiger — APPROVED** [review]
Looks good to me. Happy to merge back.

File: CHANGELOG.md [hunk 1]
```diff
@@ -156,7 +164,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 - Allow proxy to check next server - [#101](https://github.com/owncloud/user_ldap/issues/101)
 
 
-[Unreleased]: https://github.com/owncloud/user_ldap/compare/v0.15.4...master
+[Unreleased]: https://github.com/owncloud/user_ldap/compare/v0.16.0...master
+[0.15.4]: https://github.com/owncloud/user_ldap/compare/v0.15.4...v0.16.0
```
Comment (phil-davis) [inline CHANGELOG.md:168]:
```suggestion
[0.16.0]: https://github.com/owncloud/user_ldap/compare/v0.15.4...v0.16.0
```

**Review by phil-davis — COMMENTED** [review]
I will make this small change and rebase again to fix the changelog conflict.
(The branch has already been rebased since the release tag, so it won't matter to rebase again)

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/user_ldap/3732
Automatic merge failed; fix conflicts and then commit the result.

@jvillafanez I started trying to rebase locally. But it gets to a commit where it shows me conflicts for `lib/Group_LDAP.php` - that seems to have been edited in both the release branch and master in ways that are slightly different (it might just be spacing?). So I am not confident to accurately select the best/most correct code.

Please do the rebase.

**Comment by jvillafanez** [comment]
I did a rebase before the PR so maybe your branch wasn't the same as the one in the repo?

There have been conflicts in the CHANGELOG.md file, so we should recheck that everything is in place in that file in particular.

**Review by phil-davis — APPROVED** [review]
LGTM

**Pull request merged**